### DISCUSSION TITLE
shortcut: get keyboard event / actual target

### DISCUSSION THREAD

**Discussion by moklick** [vnphanquang/svelte-put#256] [Q&A] [answered]
Hey! I am using shortcut for handling several user actions. I bind all handlers on `<svelte:window />`. Now I need to filter some events. For example, if I am in an input field, I don't want to trigger the callback. Normally I would just look at the keyboard event and check if  `event.target` is an input, but that doesn't work here, because it's always the window itself. Is there any way to find out the actual target or get the keyboard event? Basically like this: https://jsbin.com/rufusujati/4/edit?html,js,console,output

**Comment by vnphanquang** [comment, marked as answer]
@moklick hmm, I see. thank you for the question.

`shortcut` is a `CustomEvent`, whose `target` is whatever node the Svelte action is placed on, so the observed behavior makes sense. For your use case we need to access the original `KeyboardEvent` object, which is not exposed in the current version.

How do you feel about an additional `originalEvent` in the `event.detail` object, something like this:

```javascript
function onShortcut(event) {
  if (event.detail.originalEvent.target.tagName === 'INPUT') {
    ...
  }
}
```

**Reply by moklick** [reply]
Hey @vnphanquang :) I think this would be a great addition for shortcut!

**Reply by vnphanquang** [reply]
@moklick now available in [@svelte-put/shortcut@3.1.0](https://github.com/vnphanquang/svelte-put/releases/tag/%40svelte-put%2Fshortcut%403.1.0). Doc page is also updated to [mention this](https://svelte-put.vnphanquang.com/docs/shortcut#original-keyboardevent). Let me know if it's working on your end.

**Reply by moklick** [reply]
wow! Thank you so much. Highly appreciated 🙏
